examples/DotNetCore/WebDemoWithEventHub/WebDemoWithEventHub/EventHubService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Azure.Identity;
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Processor;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration;
+using Microsoft.Extensions.Options;
+using System;
+using System.Threading.Tasks;
+
+namespace WebDemoWithEventHub
+{
+    public class EventHubService : ISettingsProvider
+    {
+        private Settings settings;
+
+        private IConfiguration configuration;
+
+        private EventHubConnection eventHubConnection;
+
+        private EventProcessorClient processorClient;
+
+        private IConfigurationRefresher refresher;
+
+        public EventHubService(IConfiguration config, IOptions<Settings> options)
```

Comment:
We need to figure out how to get the instance of `IConfigurationRefresher` through DI, not `IConfiguration`.

Reply:
Added another class to inject the IConfigurationRefresher.

---
In reply to: [426104383](https://github.com/Azure/AppConfiguration/pull/329#discussion_r426104383) [](ancestors = 426104383)